splink/diagnostics.py

```diff
@@ -0,0 +1,82 @@
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.session import SparkSession
+from pyspark.context import SparkContext, SparkConf
+from pyspark.sql import SQLContext
+
+from pyspark.ml.regression import LinearRegression
+from pyspark.ml.linalg import DenseVector
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.evaluation import RegressionEvaluator
+
+from pyspark.sql.types import StructType, StructField, StringType, DoubleType, Row
+import pyspark.sql.functions as f
+from pyspark.sql.functions import when
+
+
+def vif_gammas(inputdata, sparksession, sampleratio=1.0):
```

Comment:
Could we pass in the SparkSession instead of SparkContext?  This is the pattern we use elsewhere.?
https://github.com/moj-analytical-services/splink/blob/183f6986059bf434937bf121dbcd8a5cb5c48db0/splink/__init__.py#L39

Can we als put in type annotations and decorate with `@check_types` so user gets a informative error message if they pass the wrong thing

Reply:
Regarding how to use this:

aren;t we able to use the results from `get_scored_comparisons()` like below

```python
from splink import Splink

linker = Splink(settings, spark, df=df)
df_e = linker.get_scored_comparisons()
```

and then do a

`vif_gammas(df_e,spark = spark,sampleratio=0.65).show()`

Reply:
after discussion a better solution is to use 
`splink.add_gammas` for testing